pkg/apis/pipeline/v1beta1/workspace_validation.go

```diff
@@ -73,10 +73,10 @@ func (b *WorkspaceBinding) Validate(ctx context.Context) (errs *apis.FieldError)
 		return apis.ErrMissingField("projected.sources")
 	}
 
-	// The csi workspace is only supported when the alpha feature gate is enabled.
+	// The csi workspace is only supported when the beta feature gate is enabled.
 	// For a CSI to work, you must provide and have installed the driver to use.
 	if b.CSI != nil {
-		errs := version.ValidateEnabledAPIFields(ctx, "csi workspace type", config.AlphaAPIFields).ViaField("workspaces")
+		errs := version.ValidateEnabledAPIFields(ctx, "csi workspace type", config.BetaAPIFields).ViaField("workspaces")
```

Comment:
for v1beta1 apiVersion, there is no beta i.e. beta fields are "stable" per TEP-033: 
https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md

Does this mean we need to set this to stable instead of v1beta1? Basically how can we ensure this field is enabled by default in v1beta1 but not in v1? (@pritidesai @vdemeester @afrittoli @jerop @lbernick)

Reply:
we can just remove this validation check in v1beta1, no? alternatively we can modify [ValidateEnableAPIFields](https://github.com/tektoncd/pipeline/blob/933e4f667c19eaf0a18a19557f434dbabe20d063/pkg/apis/version/version_validation.go#L30) to take a crd version as well but that's a bit awkward

Reply:
Would like to confirm, by 'remove this validation check in v1beta1' means we only keep validate 'alpha' features gates?

And for the alternative solution, would it require a conversion to be implemented for feature flags of specific features?

Reply:
I just meant this line of code can be removed in v1beta1. The alternative solution shouldn't involve feature flags related to specific features.

Reply:
> For example, with `enable-api-field: stable`, one could use resolvers using the `v1beta1` API but **not** the `v1` API ? This might be slightly (or very) confusing for our users.

Thanks Vincent, can we interpret this as  we need to assure that users do understand "beta is the equivalent of stable so that "beta" features are enabled by default in v1"? If so I think this is covered in #5634 [#5634](https://github.com/tektoncd/pipeline/pull/5634/files) as:

```
The `apiVersion` field in a Tekton CRD determines whether the overall API (and its default behaviors) is considered to be in `alpha`, `beta`, or `GA`. Within a stable CRD, certain opt-in features or API fields gated may be considered `alpha` or `beta`. Similarly, within a beta CRD, certain opt-in features may be considered `alpha`. See the section on Feature Gates for details.
```

Reply:
Ok so even in v1beta1 we would need `enable-api-fields: beta` to use CSI workspace and project volume, am I right ?
@afrittoli @JeromeJu @lbernick

Reply:
@vdemeester sorry for the confusion. I think the line in this comment shall be already removed. So no `enable-api-fields: beta` needed as `beta` are the defaulted 'stable' in v1beta1.

Reply:
> @vdemeester sorry for the confusion. I think the line in this comment shall be already removed. So no `enable-api-fields: beta` needed as `beta` are the defaulted 'stable' in v1beta1.

It confuses me even more 😅. 

What is proposed here is (with the example of CSI workspace):
- To use CSI workspace:
  - in `v1` the user need to set `enable-api-fields: beta`
  - in `v1beta1` the user doesn't need to set `enable-api-fields: beta` (aka it would also work with `enable-api-fields: stable`)

am I right ?

Reply:
Yes totally!

Reply:
Just to make sure, pinging @tektoncd/core-maintainers, does that behavior make sense to every one ?
Is it inlined with https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md#promotion-to-beta-and-beyond ?

Reply:
I do find it a bit confusing but I think that is in line with the feature gates TEP

Reply:
agreed with dibyo-- it's confusing and I'd be open to changing the TEP but I think this is in line with what the TEP states.